Library/Homebrew/cmd/update-report.rb

```diff
@@ -167,16 +170,113 @@ def migrate_legacy_cache_if_necessary
     end
   end
 
-  def link_completions_and_docs
-    return if HOMEBREW_PREFIX.to_s == HOMEBREW_REPOSITORY.to_s
+  def migrate_legacy_repository_if_necessary
+    return unless HOMEBREW_PREFIX.to_s == "/usr/local"
+    return unless HOMEBREW_REPOSITORY.to_s == "/usr/local"
+    return unless ARGV.homebrew_developer?
+
+    unless HOMEBREW_PREFIX.writable_real?
+      onoe <<-EOS.undent
+        #{HOMEBREW_PREFIX} is not writable.
+
+        You should change the ownership and permissions of #{HOMEBREW_PREFIX}
+        temporarily back to your user account so we can complete the Homebrew
+        repository migration:
+          sudo chown -R $(whoami) #{HOMEBREW_PREFIX}
+      EOS
+    end
+
+    Keg::ALL_TOP_LEVEL_DIRECTORIES.each do |dir|
+      FileUtils.mkdir_p "#{HOMEBREW_PREFIX}/#{dir}"
+    end
+
+    new_homebrew_repository = Pathname.new "/usr/local/Homebrew"
+    if new_homebrew_repository.exist? && !new_homebrew_repository.children.empty?
```

Comment:
Might also need to check writability in the `new_homebrew_repository.exist?` case. If `/usr/local/Homebrew` exists, is empty, but is *not* writable, subsequent `cp_r`'s will fail.